src/styles/sidebar/components/Buttons.scss

```diff
@@ -0,0 +1,283 @@
+// visible active (pressed) state
+$IconButton-colors: map.merge(
+  $base-colors,
+  (
+    'background': transparent,
+    'hover-background': transparent,
+    'active-foreground': $color-brand,
+    'active-background': transparent,
+  )
+);
+
+$LabeledButton-colors: $base-colors;
+
+// This set of colors is light text on dark background
+$LabeledButton--primary-colors: map.merge(
+  $base-colors,
+  (
+    'foreground': $color-g1,
+    'background': $color-gmid,
+    'hover-foreground': $color-g1,
+    'hover-background': $color-g6,
+    'disabled-foreground': $color-gsemi,
+  )
+);
+
+// This set of colors is for buttons styled as links on a white background
+$LinkButton-colors: map.merge(
+  $base-colors,
+  (
+    'background': transparent,
+    'hover-background': transparent,
+    'active-background': transparent,
+    'hover-foreground': $color-link-hover,
+  )
+);
+
+// Input buttons are icon-only buttons positioned next to text inputs.
+// They have a background color (cf. standard icon buttons) and a border.
+$InputButton-colors: map.merge(
+  $base-colors,
+  (
+    'border': $color-g3,
+  )
+);
+
+// BUTTON MIXINS
+
+// Reset browser native styles. This can be used in conjunction with other
+// styles via `base` or standalone for starting a custom button's styles
+// from relative scratch
+@mixin reset {
+  @include focus.outline-on-keyboard-focus;
+  // Reset native styles
+  padding: 0;
+  margin: 0;
+  background-color: transparent;
+  border-style: none;
+}
+
+@mixin hover-state($colormap: $base-colors) {
+  &:hover:not([disabled]),
+  &:focus:not([disabled]) {
+    color: map.get($colormap, 'hover-foreground');
+    background-color: map.get($colormap, 'hover-background');
+    @content;
+  }
+  transition: color 0.2s ease-out, background-color 0.2s ease-out,
+    opacity 0.2s ease-out;
+}
+
+@mixin active-state($colormap: $base-colors) {
+  &[aria-expanded='true'],
+  &[aria-pressed='true'] {
+    color: map.get($colormap, 'active-foreground');
+    background-color: map.get($colormap, 'active-background');
+    @content;
+
+    &:hover:not([disabled]),
+    &:focus:not([disabled]) {
+      color: map.get($colormap, 'active-foreground');
+      background-color: map.get($colormap, 'active-background');
+      @content;
+    }
+  }
+}
+
+// Base mixin for all button-ish components
+@mixin base($colormap: $base-colors) {
+  @include reset;
+
+  border-radius: $border-radius;
+  border: none;
+  padding: 0.5em;
+
+  font-size: map.get($base-sizes, 'medium');
+  font-weight: 700;
+  white-space: nowrap; // Keep multi-word button labels from wrapping
+
+  color: map.get($colormap, 'foreground');
+  background-color: map.get($colormap, 'background');
+
+  &--size-small {
+    font-size: map.get($base-sizes, 'small');
+  }
+
+  &--size-large {
+    font-size: map.get($base-sizes, 'large');
+  }
+
+  // (icon) SVG sizing is relevant
+  svg {
+    width: 1.25em;
```

Comment:
Currently, we use `em` sizing for buttons, but `px` sizing for the icons within them, which is going to lead to some curious results in different combinations. The "average" button in the client has `13px` font (i.e. `1em = 13px`). This is in practice a `1:1.25` ratio `font:icon`.